src/util/console.cpp

```diff
@@ -0,0 +1,98 @@
+
+#include "console.h"
+
+#ifdef __WINDOWS__
+#include <windows.h>
+#include <io.h> // Debug Console
+typedef BOOL(WINAPI* pfSetCurrentConsoleFontEx)(HANDLE, BOOL, PCONSOLE_FONT_INFOEX);
+
+
+Console::Console() {
+
+    // Setup Windows console encoding
+    // toLocal8Bit() returns the ANSI file encoding format
+    // this does not necessarily match the OEM console encoding
+    // https://www.microsoft.com/resources/msdn/goglobal/default.mspx
+    // In case of a German Windows XP to 10 console encoding is cp850
+    // where files encoding is cp1252
+    // Qt has no solution for it https://bugreports.qt.io/browse/QTBUG-13303
+    // http://stackoverflow.com/questions/1259084/what-encoding-code-page-is-cmd-exe-using
+    // We try to change the console encoding to file encoding
+    // For a German windows we expect
+    // LOCALE_IDEFAULTANSICODEPAGE "1252" // ANSI Codepage used by Qt toLocal8Bit
+    // LOCALE_IDEFAULTCODEPAGE "850" // OEM Codepage Console
+
+    UINT m_oldCodePage;
+    bool m_shouldResetCodePage = false;
+    if(AttachConsole(ATTACH_PARENT_PROCESS)) {
+        // we are started from a console porcess
+        int fd;
+        FILE *fp;
+
+        fd = _open_osfhandle((long) GetStdHandle(STD_OUTPUT_HANDLE), 0);
+        fp = _fdopen(fd, "w");
+        *stdout = *fp;
```

Comment:
I believe you need stdio.h for stdin/stderr. It may be included from other headers though.